http/server.go

```diff
@@ -182,6 +182,15 @@ func (h *handler) PublicRand(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	h.pendingLk.RLock()
```

Comment:
I think it's totally fine with a rlock but I was just thinking that it is not necessary if you only look at the current time, so there's no need to have the http server "stateful" (and no lock).

Reply:
we need to look at the state to know if we have the state :)

Reply:
? No I'm saying that you could only look at the current time `time.Now()` and look if the request contains a round higher than the round at the current time. No state here.

Reply:
But note I have approved that PR :)